api/v1/controllers/TeamController.js

```diff
@@ -0,0 +1,105 @@
+var errors = require('../errors');
+var services = require('../services');
+var roles = require('../utils/roles');
+
+var router = require('express').Router();
+
+function joinTeam (req, res, next) {
+  var email = req.auth.email;
+  var teamName = req.body['teamName'];
```

Comment:
Do you think it's better to have the team ID to join instead? I have a feeling that this is here because the team will be auto-registered as needed, but I haven't read down far enough yet